File: HRM_systemAnguler/src/app/services/project.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { ProjectModel } from '../models/project.model';
import { Observable } from 'rxjs';
import { environment } from '../../environments/environment';
@Injectable({
  providedIn: 'root',
})
export class ProjectService {
  private apiUrl = `${environment.baseUrl}/api/project`;
  constructor(private http: HttpClient) {}
  createProject(project: ProjectModel): Observable<ProjectModel> {
    return this.http.post<ProjectModel>(this.apiUrl, project);
  }
  getAllProjects(): Observable<ProjectModel[]> {
    return this.http.get<ProjectModel[]>(this.apiUrl);
  }
  getByEmployeeId(employeeId: number): Observable<ProjectModel[]> {
    return this.http.get<ProjectModel[]>(`${this.apiUrl}/employee/${employeeId}/projects`);
  }
  getProjectById(id: number): Observable<ProjectModel> {
    return this.http.get<ProjectModel>(`${this.apiUrl}/${id}`);
  }
  updateProject(id: number, project: ProjectModel): Observable<ProjectModel> {
    return this.http.put<ProjectModel>(`${this.apiUrl}/${id}`, project);
  }
  deleteProject(id: number): Observable<string> {
    return this.http.delete<string>(`${this.apiUrl}/${id}`);
  }
}

File: HRM_system/src/main/java/com/emranhss/HRM_system/project/ProjectService.java
package com.emranhss.HRM_system.project;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface ProjectService {

    
    ProjectResponseDto createProject(ProjectRequestDto dto);

    
    List<ProjectResponseDto> getAllProjects();

    
    ProjectResponseDto getProjectById(Long id);

    
    ProjectResponseDto updateProject(Long id,
                                     ProjectRequestDto dto);

    
    void deleteProject(Long id);


    List<ProjectResponseDto>getByEmployeeId(Long id);

}

